components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -963,6 +963,30 @@ else if (segment instanceof Attachment) {
       }
       indices.add(i);
       indexIntoPlanes.put(c, indices);
+      //Add series metadata : populate position list
+      for (DimensionEntry dimension : p.directoryEntry.dimensionEntries) {
+          if (dimension == null) {
+            continue;
+          }
+          switch (dimension.dimension.charAt(0)) {
+              case 'S':
+                  setSeries(p.coreIndex);
```

Comment:
This should likely be ```setCoreIndex(p.coreIndex)```, otherwise the indexing may be wrong when working with files that contain pyramids.